### PR TITLE
[CopyFilesOverSSHV0] Fixed bug with UNC target paths

### PR DESCRIPTION
**Pull request by egor-bryzgalov**
**Task name**: CopyFilesOverSSHV0

**Description**: Fixed bug with UNC target paths. UNC paths are going to be handled like absolute paths. Added unit tests. Added debug logging.

**Documentation changes required:**No

**Added unit tests:** Yes

**Attached related issue:** #13477 

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Manually tested applied changes
- [x] e2e tests were passed successfully

### CODE REVIEW

**Review by sdobrodeev — APPROVED** [review]
LGTM, thanks!

File: Tasks/CopyFilesOverSSHV0/utils.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    }
+    return filePath;
+}
+
+/**
+ * Determines whether {path} is an UNC path.
+ * @param path 
+ */
+export function pathIsUNC(path: string): boolean {
+    const regExp: RegExp = /^[\\]{2,}[^\\\/]+[\\\/]+[^\\\/]+/;
```
Comment (anatolybolshakov) [inline Tasks/CopyFilesOverSSHV0/utils.ts:21]: Could you please move this declaration to some constants - at least out of method?

File: Tasks/CopyFilesOverSSHV0/Tests/L0.ts [hunk 2]
```diff
@@ -9,8 +10,32 @@ describe('CopyFilesOverSSHV0 Suite', function () {
...
+        it('Should not recognize strings as UNC paths', function (done: MochaDone) {
+            const paths: string[] = [
+                '//host\\one\\two',
+                '//host\\one\\two\\',
+                '//host\\one\\two/file',
+                '//host/one/two/file',
+                '\\host\\one\\two',
+            ];
+            for (const path of paths) {
+                assert(!utils.pathIsUNC(path), `Should be recognized as UNC path: ${path}`);
```
Comment (anatolybolshakov) [inline Tasks/CopyFilesOverSSHV0/Tests/L0.ts:36]: Typo: 'Should not be recognized'?

File: Tasks/CopyFilesOverSSHV0/utils.ts (hunk 1 above)
Comment (egor-bryzgalov) [reply Tasks/CopyFilesOverSSHV0/utils.ts]: Thanks for review point. Moved the declaration.

File: Tasks/CopyFilesOverSSHV0/utils.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    return UNCPathPattern.test(path);
+}
+
+/**
+ * Gets OS specific command to clean folder in specified path.
+ * @returns {string} OS specific command to clean target folder on the remote machine
+ * @param {string} targetFolder path to target folder
+ */
+export function getCleanTargetFolderCmd(targetFolder: string): string {
+    const isWindowsOnTarget: boolean = tl.getBoolInput('isWindowsOnTarget', false);
```
Comment (anatolybolshakov) [inline Tasks/CopyFilesOverSSHV0/utils.ts:32]: Nit: could you please move tl.getBoolInput our of utils method?

File: Tasks/CopyFilesOverSSHV0/Tests/L0.ts [hunk 4]
```diff
@@ -9,8 +10,32 @@ describe('CopyFilesOverSSHV0 Suite', function () {
     after(() => {
     });
 
-    it('Does a basic hello world test', function(done: MochaDone) {
-        // TODO - add real tests
-        done();
+    context('Utils tests: ', function () {
+        it('Should recognize UNC paths', function (done: MochaDone) {
+            const paths: string[] = [
```
Comment (anatolybolshakov) [inline Tasks/CopyFilesOverSSHV0/Tests/L0.ts:15]: Could you please move path definitions out of L0.ts module? Maybe to some separate module for unils tests?

**Review by anatolybolshakov — APPROVED** [review]

**Review by damccorm — APPROVED** [review]

**Pull request merged**